### PR TITLE
update go4.org/unsafe/assume-no-moving-gc for panic on dev systems

### PR DESCRIPTION
**Pull request by jacobsalmela**
# Summary and Scope

- Fixes this error seen on local dev systems when running `make test`

```
k  	github.com/Cray-HPE/cani/internal/inventory	0.161s	coverage: 21.3% of statements
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.21 runtime. You can try updating this package by running "go get go4.org/unsafe/assume-no-moving-gc@latest". If that doesn't work and you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH="go1.21" set. Notably, if go1.21 adds a moving garbage collector, this program is unsafe to use.

goroutine 1 [running]:
go4.org/unsafe/assume-no-moving-gc.init.0()
	/root/jsalmela/cani/vendor/go4.org/unsafe/assume-no-moving-gc/untested.go:25 +0x1da
FAIL	github.com/Cray-HPE/cani/internal/provider/csm	0.011s
panic: Something in this program imports go4.org/unsafe/assume-no-moving-gc to declare that it assumes a non-moving garbage collector, but your version of go4.org/unsafe/assume-no-moving-gc hasn't been updated to assert that it's safe against the go1.21 runtime. You can try updating this package by running "go get go4.org/unsafe/assume-no-moving-gc@latest". If that doesn't work and you want to risk it, run with environment variable ASSUME_NO_MOVING_GC_UNSAFE_RISK_IT_WITH="go1.21" set. Notably, if go1.21 adds a moving garbage collector, this program is unsafe to use.

goroutine 1 [running]:
go4.org/unsafe/assume-no-moving-gc.init.0()
	/root/jsalmela/cani/vendor/go4.org/unsafe/assume-no-moving-gc/untested.go:25 +0x1da
FAIL	github.com/Cray-HPE/cani/internal/provider/csm/ipam	0.011s
ok  	github.com/Cray-HPE/cani/internal/provider/csm/sls	0.044s	coverage: 6.4% of statements
ok  	github.com/Cray-HPE/cani/internal/provider/csm/validate	0.156s	coverage: 50.7% of statements
ok  	github.com/Cray-HPE/cani/internal/provider/csm/validate/checks	0.020s	coverage: 40.0% of statements
ok  	github.com/Cray-HPE/cani/internal/provider/csm/validate/common	0.013s	coverage: 8.3% of statements
FAIL
make: *** [Makefile:143: unittest] Error 1
```

# Risks and Mitigations
 

Low

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -139,6 +139,14 @@ shellspec:
 validate-hardware-type-schemas:
 	go run ./pkg/hardwaretypes/validate pkg/hardwaretypes/hardware-types/schema  pkg/hardwaretypes/hardware-types/
 
+sim:
```
Comment (shunr-hpe) [inline Makefile:142]: The name could be: `sim-install-and-setup` or `sim-setup`

File: Makefile [hunk 2]
```diff
@@ -139,6 +139,14 @@ shellspec:
...
+	if ! [ -d hms-simulation-environment ]; then git clone https://github.com/Cray-HPE/hms-simulation-environment.git; fi
+	spec/support/bin/setup_simulator.sh ./hms-simulation-environment ../testdata/fixtures/sls/no_hardware.json
+
+spec:
```
Comment (shunr-hpe) [inline Makefile:146]: The target name could be: `shellspec-install` or `spec-install`

File: Makefile (hunk 1 above)
Comment (shunr-hpe) [reply Makefile:142]: or `sim-start`

Comment (jacobsalmela) [reply Makefile]: Wilco.  Breaking this off into its own PR.  This one will simply allow the tests to run without a panic on the NCNs

File: Makefile (hunk 2 above)
Comment (jacobsalmela) [reply Makefile]: Wilco.  Breaking this off into its own PR.  This one will simply allow the tests to run without a panic on the NCNs

**Review by shunr-hpe — APPROVED** [review]

**Pull request merged**